### PR TITLE
feat: add `svelte/no-restricted-html-elements` rule

### PR DESCRIPTION
**Pull request by baseballyama**
close: https://github.com/sveltejs/eslint-plugin-svelte/issues/436

### CODE REVIEW

File: docs/rules/no-restricted-html-elements.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+## :wrench: Options
+
+This rule takes a list of strings, where each string is an HTML element name to be restricted:
+
+```json
+{
+  "svelte/no-restricted-html-elements": [
+    "error",
+    ["h1", "h2", "h3", "h4", "h5", "h6"]
```
Comment (ota-meshi) [inline docs/rules/no-restricted-html-elements.md:76]: I prefer flattening over using nested arrays here.
What do you think about flattening?

File: src/rules/no-restricted-html-elements.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { createRule } from "../utils"
+
+export default createRule("no-restricted-html-elements", {
+  meta: {
+    docs: {
+      description: "disallow specific HTML elements",
+      category: "Extension Rules",
```
Comment (ota-meshi) [inline src/rules/no-restricted-html-elements.ts:7]: Hmm. I think the better category for this rule is `Stylistic Issues` among the categories we currently have, but we may need to create a new category 🤔

File: docs/rules/no-restricted-html-elements.md (hunk 1 above)
Comment (baseballyama) [reply docs/rules/no-restricted-html-elements.md]: I forgot to update this. I updated!

File: src/rules/no-restricted-html-elements.ts (hunk 2 above)
Comment (baseballyama) [reply src/rules/no-restricted-html-elements.ts]: Ah, I misunderstood the meaning of `Extension Rules`.
Just now I changed the category.
Should we refactor categories in the next major version?

Comment (ota-meshi) [reply src/rules/no-restricted-html-elements.ts]: It looks like we need to run `pnpm run update`.

Comment (ota-meshi) [reply src/rules/no-restricted-html-elements.ts]: > Should we refactor categories in the next major version?

We can always make changes if we have ideas for categorization that make it easier for users to understand 😉

File: docs/rules/no-restricted-html-elements.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+```
+
+Alternatively, the rule also accepts objects.
+
+```json
+{
+  "svelte/no-restricted-html-elements": [
+    "error",
+    {
+      "elements": "h1", "h2", "h3", "h4", "h5", "h6",
```
Comment (ota-meshi) [inline docs/rules/no-restricted-html-elements.md:93]:
```suggestion
      "elements": ["h1", "h2", "h3", "h4", "h5", "h6"],
```

Comment (baseballyama) [reply docs/rules/no-restricted-html-elements.md]: 🙏

File: src/rules/no-restricted-html-elements.ts (hunk 2 above)
Comment (baseballyama) [reply src/rules/no-restricted-html-elements.ts]: I thought about it a bit, but I think `Stylistic Issues` is fine for now.

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**